### ISSUE TITLE
"Collision via traces" still uses collision groups for particles

### ISSUE THREAD

**Issue by yuberee** [sboxgame/issues#2985]
### Describe the bug

Right now, your particles can only collide with the world, and not entities.
You can't change it right now because the only option given in the "Collision via traces" constraint is for collision groups, which have been replaced, rendering it useless.

### To Reproduce

1. Open/Create a particle which uses the "Collision via traces" constraint
2. Find the "Collision group" property, modify it to anything
3. Particles will always only collide with the world

### Expected behavior

There should be an extra property to define which collision tag the particle uses, so it can check your project's Collision Matrix during physics calculations.

### Media/Files

![image](https://user-images.githubusercontent.com/59583743/222925198-225709d2-2bb8-4f62-82d3-a58194b79208.png)

**Comment by aylaylay** [comment]
Particles have always only collided with static objects, never movable entities. I can remove the collision group shit but I dunno how useful adding tags would be.

**Comment by aylaylay** [comment]
Actually, I might be talking shit, I'll see what I can do

**Comment by aylaylay** [comment]
Just pushed this, I don't have a particle to test with so let me know if it works

**Comment by aylaylay** [comment]
![sbox-dev_ow8247A8Tv](https://user-images.githubusercontent.com/1667289/232410699-4d67b17f-5892-4714-ab70-d9839c1b6bc8.png)

**Comment by yuberee** [comment]
Yup I can confirm adding "solid" will make it collide with anything with the "solid" tag, including entities.
I haven't tried throwing in tags in the "Exclude" but I'm sure it works.

**Comment by Xenthio** [comment]
i'm finding that exclude tags seem to not work?
![image](https://user-images.githubusercontent.com/28588188/232637582-617832bc-743b-45a8-8686-ceac56a74245.png)
doing this seems to make it not collide with anything

**Comment by aylaylay** [comment]
I'll have a look tomorrow.

**Comment by Xenthio** [comment]
actually it seems particle collisions aren't working at all for me,

**Comment by aylaylay** [comment]
Small example would be good if possible

**Comment by Xenthio** [comment]
I can't make one, It seems like it's random sometimes it works, sometimes it kinda works, other times it doesn't work at all

**Comment by Xenthio** [comment]
okay i've pinpointed an issue causing my problem, it seems that if i dont have a sprite renderer the collisions don't work properly, i'm using a projected texture renderer and things fall through the map, but as soon as i add a sprite everything just seems to work fine

**Comment by aylaylay** [comment]
Okay I'll have a look through, most likely missed something.

**Comment by aylaylay** [comment]
I've pushed something, hopefully it should be working now. Tags were unitialized so that might explain why it sometimes worked and sometimes didn't.
